File: 650/Solution.go
func minSteps(n int) int {
    return count(n, 0)
}

func count(n , res int) int{
    if n == 1 { return res }
    x := n / 2
    for n % x != 0 { x-- }
    res += n / x
    return count(x , res)
}

File: 650/Solution.java
class Solution {
    int res = 0;
    public int minSteps(int n) {
        if(n == 1 ) return res;
        int x = n / 2;
        while(n % x != 0) x--;
        res += n / x;
        return minSteps(x);
    }
}

// time O(n) - O(log n)
// space O(1)
